File: scala-simple-http-server/src/main/scala/simplehttpserver/Status.scala
package simplehttpserver

/** HTTPレスポンスのステータス */
sealed abstract class Status(val value: String)

object Status {
  case object Ok extends Status("200 OK")
  case object BadRequest extends Status("400 BadRequest")
  case object Forbidden extends Status("403 Forbidden")
  case object NotFound extends Status("404 NotFound")
}

File: java-simple-http-server/src/main/java/simplehttpserver/enums/Status.java
package simplehttpserver.enums;

public enum Status {
    OK("200 OK"),
    BAD_REQUEST("400 BadRequest"),
    FORBIDDEN("403 Forbidden"),
    NOT_FOUND("404 NotFound"),
    INTERNAL_SERVER_ERROR("500 InternalServerError");

    public final String statusCode;

    Status(String statusCode) {
        this.statusCode = statusCode;
    }
}
